### DISCUSSION TITLE
Idea about Ballast Flora detection and more

### DISCUSSION THREAD

**Discussion by ech0-v1** [FakeFishGames/Barotrauma#15310] [Ideas and suggestions]
Maybe Sonar can be upgradeable like 10.000 credits. By so even same scan diameter distance it can detect Ballast Flora Spores much far away and clearer. By upgrade that can be send much more powerful frequency that makes Sonar, like Real Sonar mod alike. Not that much violent like the mod applies but upgraded sonar can be damage giving **affliction** as a sonar blast. Diving suit's can have full resistance to that external damage. And at Heer HUD, there can be a switch which is VHFS, Very High Frequency Sonar mode. Can be switchable alongside when Active Sonar mode is already on. Because of VHFS, ship's navigation terminal should use much more load on electrical energy. When piezo crystals electromagnetics hits sub that can interfere or jam VHFS also.

**Comment by jigardi** [comment]
Have my upvote.

I'm not sure I agree with everything in this suggestion;  but I definitely think that sonar is an extremely underdeveloped part of the game, and therefore has a lot of room for improvement.

**Reply by NoahRoopull** [reply]
Even a sonar range upgrade would be nice.
